Automattic/woocommerce-payments
# Inbox notifications on overview screen

Fixes #1662 

#### Changes proposed in this Pull Request

Adds a InboxPanel component to the overview screen that display's woocommerce payment notes.

This adds a InboxPanel component within the overview directory to be displayed on the overview screen.
In order for us to only query `woocommerce-payments` notes, I also had to add a few filters to the `class-wc-payments.php`, which allows us to query by `source`.
I also added a helper function for updating the woocommerce_meta field in the current user object, this is used for showing if a note is new or not - storing the timestamp in the meta field. We do the same in **WooCommerce > Home**.



![wcpay-notes](https://user-images.githubusercontent.com/2240960/120848813-a5370e80-c54b-11eb-98c6-43f0fe660132.gif)


#### Testing instructions
- Load this branch
- Go to **Payments > Overview**
- It should briefly show a loading inbox note placeholder
- Two notes should be present - these are the only woocommerce-payments notes (at-least for me) - `Set up refund policy` & `Force secure checkout` (as shown in the gif above) 
- All note interactions should work (dismissing one/all, and clicking on the action).
- With the `debug` field set to `wc-admin:*` in localstorage you should see the `wcpay_inbox_note_view` when scrolling down for each note.





*

-------------------
